mono/metadata/sgen-mono.c

```diff
@@ -1643,6 +1643,116 @@ report_handle_stack_roots (GCRootReport *report, SgenThreadInfo *info, gboolean
 	mono_handle_stack_scan (info->client_info.info.handle_stack, report_handle_stack_root, &ud, ud.precise, FALSE);
 }
 
+static void*
+get_aligned_stack_start (SgenThreadInfo *info)
+{
+	void* aligned_stack_start = (void*)(mword) ALIGN_TO ((mword)info->client_info.stack_start, SIZEOF_VOID_P);
```

Comment:
This alignment is debatable.
1. On amd64 it will already be pointer aligned.
2. Are there systems where stack might not be pointer-aligned?
3. It is not correct for grow-up stack (HPPA).